sycl/test-e2e/ESIMD/config_2d_mem_access_test.cpp

```diff
@@ -47,15 +47,17 @@ int main() {
       payload.set_surface_pitch(SurfacePitch);
       payload.set_x(x);
       payload.set_y(y);
-      result[0] = payload.get_surface_width();
-      result[1] = payload.get_surface_height();
-      result[2] = payload.get_surface_pitch();
-      result[3] = payload.get_x();
-      result[4] = payload.get_y();
-      result[5] = payload.get_width();
-      result[6] = payload.get_height();
-      result[7] = payload.get_number_of_blocks();
-      auto p = payload.get_data_pointer();
+      config_2d_mem_access<uint32_t, BlockWidth, BlockHeight, NumBlocks>
+          payloadResult(payload);
```

Comment:
Why is this new object needed? Is it added here only to test the constructor? If Yes, then it would be better to have it somewhere in compile-only LIT test that is run always, not only on PVC.